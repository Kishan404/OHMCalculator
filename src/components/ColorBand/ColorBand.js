import React, { Component } from 'react';
import PropTypes from 'prop-types';

class ColorBand extends Component {
    constructor(props) {
        super(props);
        this.state = {
            selectedcolor: props.bandOptions[0][0].toLowerCase()
        }
        this.handleColorPick = this.handleColorPick.bind(this);
    }
    /**
     * 
     * @param {event} e 
     */
    handleColorPick(e) {
    
      let pickedcolor = e.target.value;
      this.props.handleColorPick(pickedcolor,this.props.bandIndex);
      console.log('pickedcolor:'+pickedcolor);
      this.setState(()=>({selectedcolor:pickedcolor.toLowerCase()}))

    }
    /**
     * returns number in k M G format
     * @param {number} num 
     */
    formatNumtokMG(num) {
        if (num >= 1000000000) {
           return (num / 1000000000).toFixed(1).replace(/\.0$/, '') + 'G';
        }
        if (num >= 1000000) {
           return (num / 1000000).toFixed(1).replace(/\.0$/, '') + 'M';
        }
        if (num >= 1000) {
           return (num / 1000).toFixed(1).replace(/\.0$/, '') + 'k';
        }
        return num;
   }
   /**
    * Formate color option bansed on the band Type
    * @param {*} bandOption 
    */
    formatColorOption(bandOption) {
        
        switch(this.props.bandIndex) {
            case 0:
                return bandOption[0]+'  ('+bandOption[1]+')';  
                break;
            case 1:
                return bandOption[0]+'  ('+bandOption[1]+')';
                break;
            case 2:
                return bandOption[0]+'  ('+this.formatNumtokMG(+bandOption[1])+')';
                break;
            case 3:
                return bandOption[0]+'  (Â±'+bandOption[1]+'%)';
                break;    
            default:
        }

    }

    render() {
        return (
            <div className="colorband">
                <label className="label" htmlFor={'band'+this.props.bandIndex}>{this.props.bandLabel}</label> 
                <select className={`colorselector ${this.state.selectedcolor}`} id={'band'+this.props.bandIndex} onInput={this.handleColorPick}>
                   {this.props.bandOptions.map((bandOption,key)=>{ return <option key={key} className={`coloroption ${bandOption[0].toLowerCase()}`}value={bandOption[0]} >{this.formatColorOption(bandOption)}</option>})}
                </select>
            </div>
        );
    }
}

ColorBand.propTypes = {
bandIndex : PropTypes.number,
bandLabel : PropTypes.string,
bandOptions : PropTypes.array,
handleColorPick : PropTypes.func
}

export default ColorBand;