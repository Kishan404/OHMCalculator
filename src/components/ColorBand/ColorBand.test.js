import React, { Component } from 'react';
const ReactTestRenderer = require('react-test-renderer');
import ColorBand from './ColorBand';



describe('Colorband Snapshot test', () => {
    it('Should compare the component with a snapshot', () => {
      const component = ReactTestRenderer.create(
        <ColorBand 
          key ={0}
          bandIndex = {0}
          bandLabel = {'Band A'} 
          bandOptions = {['gray','Brown']} />
      );
      const json = component.toJSON();
      expect(json).toMatchSnapshot();
    })
  });
