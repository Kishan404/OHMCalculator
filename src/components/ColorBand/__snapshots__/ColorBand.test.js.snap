// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Colorband Snapshot test Should compare the component with a snapshot 1`] = `
<div
  className="colorband"
>
  <label
    className="label"
    htmlFor="band0"
  >
    Band A
  </label>
  <select
    className="colorselector g"
    id="band0"
    onInput={[Function]}
  >
    <option
      className="coloroption g"
      value="g"
    >
      g  (r)
    </option>
    <option
      className="coloroption b"
      value="B"
    >
      B  (r)
    </option>
  </select>
</div>
`;
